### PR TITLE
Search component for products search by filtering through includes me…

### PR DESCRIPTION
**Pull request by almozaai**
…thod

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -29,6 +29,10 @@
           margin: 0.5rem;
           width: calc(100%/4);
         }
+        #search{
```
Comment (nikkiconsolacion) [inline index.html]: i think it needs to be .search instead of #search since you're using className instead of id

File: src/Search.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Products from './Products';
+
+class _Search extends React.Component{
+    constructor(){
+        super()
+        this.state =  {
+            searchItems: [],
+            inpu: ''
```
Comment (nikkiconsolacion) [inline src/Search.js]: spelling error on line 10. should be 'input'

File: src/Search.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+         
+         this.setState({ searchItems, input: e.target.value })
+    }
+
+    render(){
+       const { searchItems, input } = this.state
+
+   return( 
+       <div>
+           <form>
```
Comment (nikkiconsolacion) [inline src/Search.js]: do we need an onSubmit for the input form?

File: src/Search.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                <input type="text" name="search" id="search" onKeyUp={e => this.startSearch(e)}/>
+            </form>
+            <div className='ordering' >
+          {searchItems.length === 0 && !input ? <Products />:
+            searchItems.map(product => <div key={product.id} className='border' >
+              {product.name}
+              <br/>
+              ${product.price}
+              <br/>
+              <button onClick={() => this.props.toCreate(product)} >Add to Cart</button>
```
Comment (nikkiconsolacion) [inline src/Search.js]: need to add the toCreate: (item) => dispatch(addCartItem(item)) to the mapStateToDispatch

File: src/store.js [hunk 5]
```diff
@@ -72,6 +76,16 @@ const getProductsThunk = ()=>{
   }
 }
 
+const addToCart = (item) => {
+  return async (dispatch) => {
+    const respoanse = (await axios.post(`/api/products/${item.id}`, item)).data;
```
Comment (nikkiconsolacion) [inline src/store.js]: spelling error 'response' instead of 'respoanse' on line 81 and 82

File: src/store.js [hunk 6]
```diff
@@ -72,6 +76,16 @@ const getProductsThunk = ()=>{
   }
 }
 
+const addToCart = (item) => {
```
Comment (nikkiconsolacion) [inline src/store.js]: Sorry to be picky, can we name this 'addToCartThunk'? Just to be consistent with the rest of the thunks. Can then change the action creator '_addToCart' to either 'addToCart' or 'addCartItem' just to be consistent. But then don't forget to change the export on line 91 as well.

File: src/index.js [hunk 7]
```diff
@@ -23,8 +24,10 @@ class _App extends Component{
     return (
       <HashRouter>
         <Route component={Nav} />
+        
         <Route path='/' component={Home} exact />
-        <Route path='/users' component={Users} />
+        <Route exact path='/users' component={Users} />
+        <Route path='/users/:id' component={User} />
```
Comment (nikkiconsolacion) [inline src/index.js]: looks great! we just might get an error 'cannot find module User' until we build the User.js component

File: src/Search.js (hunk 2 above)
Comment (almozaai) [reply src/Search.js]: fixed

File: index.html (hunk 1 above)
Comment (almozaai) [reply index.html]: sorry I didn't see this comment, I did not fix in last pull request

File: src/Search.js (hunk 3 above)
Comment (almozaai) [reply src/Search.js]: no need because of no button there.

File: src/Search.js (hunk 4 above)
Comment (almozaai) [reply src/Search.js]: I just copy Product component code by Mathew to safe time.
and we need it later when fire dispatch(attToCart(item))

File: src/store.js (hunk 5 above)
Comment (almozaai) [reply src/store.js]: I am sorry I will fix it later I am at duty now

File: src/index.js (hunk 7 above)
Comment (almozaai) [reply src/index.js]: Yes, I have many ideas in my head one of it how to add items to the user, I thought I can finish it before 3 am but I facing some issues during work and take time to fix it, and leave some behind and you catch it :) Thank you Nikki

File: src/store.js (hunk 6 above)
Comment (almozaai) [reply src/store.js]: Ok, I appreciate your effort.

**Pull request closed without merging**